hera_cal/smooth_cal.py

```diff
@@ -1000,8 +1033,21 @@ def time_freq_2D_filter(self, freq_scale=10.0, time_scale=1800.0, tol=1e-09, fil
                 filtered, info = time_freq_2D_filter(gain_grid, wgts_grid, self.freqs, self.time_grid, freq_scale=freq_scale,
                                                      time_scale=time_scale, tol=tol, filter_mode=filter_mode, maxiter=maxiter,
                                                      window=window, dpss_vectors=dpss_vectors, eigenval_cutoff=eigenval_cutoff,
-                                                     method=method, fit_method=fit_method, cached_input=cached_input, 
-                                                     skip_flagged_edges=skip_flagged_edges, **win_kwargs)
+                                                     method=method, fit_method=fit_method, cached_input=cached_input,
+                                                     skip_flagged_edges=skip_flagged_edges, fix_phase_flips=fix_phase_flips, **win_kwargs)
+                flipped = (info['phase_flips'] == -1)
+                if np.any(flipped):
+                    print(f'{np.sum(flipped)} phase-flipped integrations detected on antenna {ant} between {self.time_grid[flipped][0]} and {self.time_grid[flipped][-1]}.')
+                    # apply flags before and after phase flips
+                    most_recent_unflagged_tind = 0
+                    if flag_phase_flip_ints:
```

Comment:
If I'm remembering correctly, I think there were some discussions in the general meeting about whether we should completely flag antennas where we find phase flips. Would it be worth adding another option to completely flag an antenna if you find a phase flip?

Reply:
Great idea! Done.